### PR TITLE
Split Keysanity Options

### LINKED ISSUE #179: Split keysanity in multiple options
Currently keysanity is either off or on. Introduce options to have keysanity enabled for one game or the other. I originally came up with the idea so that people could play with the SM keys to potentially balance the two games a bit more and require people to go back and forth rather than the vast majority of seeds almost ending up with you doing 75% of SM in one go.

### PR DESCRIPTION
**Pull request by MattEqualsCoder**
Closes #179 

This implements both keysanity options as well as item placement options.

Kinda irrelevant to this, but I also consolidated some of the good/junk/progression logic as I noticed it was kind of duplicated or spread out into different places. This opens the door up for #122 as there are technically now 3 lists, but it still doesn't fully solve all of the problems. We'd need to build things up a bit more to only register the first two swords as a progression item and not a nice to have as an example, and we'd need separate hint messages for progression vs nice to have.

### CODE REVIEW

**Comment by MattEqualsCoder** [comment]
Opening as draft until the other PR is closed, then I'll address any merge conflicts and open it as a full PR.

**Comment by MattEqualsCoder** [comment]
Merged plando changes in and addressed all issues.

File: src/Randomizer.SMZ3.Tracking/Configuration/ConfigTypes/ItemData.cs [hunk 1]
```diff
@@ -318,15 +318,47 @@ public bool TryGetTrackingResponse([NotNullWhen(true)] out SchrodingersString? r
...
+            if (InternalItemType == ItemType.Nothing || InternalItemType.IsInAnyCategory(new[] { ItemCategory.Junk, ItemCategory.Scam, ItemCategory.NonRandomized, ItemCategory.Map, ItemCategory.Compass, ItemCategory.Nice }))
+                return false;
+
+            if (InternalItemType.IsInAnyCategory(new[] { ItemCategory.SmallKey, ItemCategory.BigKey }))
+                return config?.ZeldaKeysanity == true;
+
+            if (InternalItemType.IsInCategory(ItemCategory.Keycard))
+                return config?.MetroidKeysanity == true;
+
+            // Todo: We can add special logic like checking if it's one of the first two swords
```
Comment (Vivelin) [inline src/Randomizer.SMZ3.Tracking/Configuration/ConfigTypes/ItemData.cs:359]: If we want to make the "item worth" checks any more complex we should probably move it into ItemService or something so we can do more than simple static checking.

**Review by Vivelin — APPROVED** [review]

File: src/Randomizer.SMZ3.Tracking/Configuration/ConfigTypes/ItemData.cs (hunk 1 above)
Comment (MattEqualsCoder) [reply src/Randomizer.SMZ3.Tracking/Configuration/ConfigTypes/ItemData.cs:359]: Yeah, that's fair. I think there are also a few other places that have similar logic that would be good to consolidate as well. I'll make a note in the issue for that.

**Pull request merged**